src/redux/stores/indexer/indexerApi.ts

```diff
@@ -0,0 +1,59 @@
+import { Protocols } from '@airswap/constants';
+import { RegistryV4 } from '@airswap/libraries';
+import { FullOrder, IndexedOrder, OrderFilter } from '@airswap/types';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { providers } from 'ethers';
+
+import { INDEXER_ORDER_RESPONSE_TIME_MS } from '../../../constants/configParams';
+import { AppDispatch, RootState } from '../../store';
+import { getServers } from './indexerHelpers';
+
+interface InitializeParams {
+  chainId: number,
+  provider: providers.Web3Provider,
+}
+
+export const initialize = createAsyncThunk<string[], InitializeParams>(
+  'indexer/initialize',
+  async ({ chainId, provider }) => RegistryV4.getServerURLs(
+    provider,
+    chainId,
+    Protocols.Storage,
+  ),
+);
+
+export const getFilteredOrders = createAsyncThunk<
+FullOrder[],
+{ filter: OrderFilter },
+{
+  dispatch: AppDispatch;
+  state: RootState;
+}
+>('indexer/getFilteredOrders', async ({ filter }, { getState }) => {
+  const { indexer: indexerState } = getState();
+  let orders: Record<string, IndexedOrder<FullOrder>> = {};
+  if (!indexerState.isInitialized) return [];
+
+  const servers = await getServers(indexerState.urls);
+
+  const orderPromises = servers.map(async (server) => {
+    try {
+      const orderResponse = await server.getOrders(filter);
+      const ordersToAdd = orderResponse.orders;
+      orders = { ...orders, ...ordersToAdd };
```

Comment:
When writing a function try to apply "sequential execution" or "top-down execution".

Sequential execution means that the code should be organised in a way that the line of logic progresses from top to bottom, without modifying variables declared above. This approach enhances code readability and maintainability by reducing complexity and making it easier to understand the flow of execution.

Here you're modifying `orders` that's defined a few lines above. You should write your code in a more linear way so it's easier to follow.

It's a bit complicated because you're using `Promise.race`. So I did a rewrite of this function.